pkg/controller/controller_instance_test.go

```diff
@@ -215,7 +215,63 @@ func TestReconcileInstanceNonExistentServicePlan(t *testing.T) {
 	}
 }
 
-func TestReconcileInstanceWithParameters(t *testing.T) {
+func TestReconcileInstanceWithInvalidParameters(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeBrokerClient, testController, sharedInformers := newTestController(t, noFakeActions())
+
+	sharedInformers.Brokers().Informer().GetStore().Add(getTestBroker())
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+
+	instance := getTestInstance()
+
+	// Secret with invalid JSON
+	invalidSecret := &v1.Secret{
+		Data: map[string][]byte{
+			testSecretKey: []byte("invalid json"),
```

Comment:
Ok, going to implement this today.
Hopefully this will be the last change for this PR (and all other improvements are to be addressed through separate PRs)